javaslang/src/main/java/javaslang/control/Try.java

```diff
@@ -39,6 +45,39 @@
         }
     }
 
+
+    /**
+     * Creates a Try of a Supplier.
+     *
+     * @param supplier A supplier
+     * @param <T>      Component type
+     * @return {@code Success(supplier.get())} if no exception occurs, otherwise {@code Failure(throwable)} if an
+     * exception occurs calling {@code supplier.get()}.
+     */
+    static <T> Try<T> ofSupplier(Supplier<? extends T> supplier) {
+        try {
```

Comment:
Let's return `of(supplier::get)` to reduce duplicate code. The compiler will translate the method reference to a simple method call - there's no additional overhead.

Please do this for all other static factory methods we introduce in this PR, too.